plugins/updater/src/updater.rs

```diff
@@ -540,12 +540,71 @@ impl Update {
         // shouldn't drop but we should be able to pass the reference so we can drop it once the installation
         // is done, otherwise we have a huge memory leak.
 
-        let tmp_dir = tempfile::Builder::new().tempdir()?.into_path();
-        let archive = Cursor::new(bytes);
-        let mut extractor = zip::ZipArchive::new(archive)?;
-        extractor.extract(&tmp_dir)?;
+        // we support 2 type of files exe & msi for now
+        // If it's an `exe` we expect an NSIS installer.
+        enum UpdaterType {
+            Nsis {
+                path: PathBuf,
+                // For extending temp file's life time
+                #[allow(unused)]
+                temp_path: Option<tempfile::TempPath>,
+            },
+            Msi {
+                path: PathBuf,
+                // For extending temp file's life time
+                #[allow(unused)]
+                temp_path: Option<tempfile::TempPath>,
+            },
+        }
 
-        let paths = fs::read_dir(&tmp_dir)?;
+        let updater = 'updater: {
```

Comment:
instead of using a labeled block with a break statement, we could split these 3 conditions into 2 functions

```rs
enum UpdaterType {
  Msi,
  Nsis
}

let (updater_type, path, _temp) = if is_zip() {
  Self::extract_zip(&bytes)?
} else {
  Self::extract_bin(&bytes)?
};
``` 

where `extract_bin` and `extract_zip` will check if msi or nsis and write with appropriate file extension then returns whether this update package is MSI or not

```rs
fn extract_zip(bytes: &[u8]) -> Result<(UpdaterType, PathBuf, Option<TempPath>)>;
```